packages/dev/core/src/Misc/deepCopier.ts

```diff
@@ -13,6 +13,8 @@ const CloneValue = (source: any, destinationObject: any) => {
         return source.clone(destinationObject);
     } else if (source.clone) {
         return source.clone();
+    } else if (Array.isArray(source)) {
+        return source.slice();

```

Comment:
I would disagree here because the array content will NOT be deepCopied in that case

Reply:
The deep copy of an array happen in here: https://github.com/BabylonJS/Babylon.js/pull/13840/files#diff-b958f318a08b9b7898b53cea2dcaddf0bb809767dd6c36c1d750791fb93361b3L75

Reply:
Shouldn't we slowly switch to `structuredClone`?

Reply:
Maybe but this is a different topic ;)

Reply:
(https://github.com/BabylonJS/Babylon.js/pull/13840/files#diff-b958f318a08b9b7898b53cea2dcaddf0bb809767dd6c36c1d750791fb93361b3L75)

But clones only primitive values. If it's an array of arrays you need to take care of it in the `CloneValue` function. Isn't this that case?

Reply:
It will call clone on the items of the array. We did not support array of arrays as it was unnecessary

Reply:
Oh wait! I see what you mean! You needed it actually in your case

Reply:
ok all good for me!